test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,142 @@
+}
+
+template<Alphabet alphabet_t>
+void mutate_deletion(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed = 0){
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(seq) - overlap);
+    seq.erase(seq.begin() + random_pos(gen));
+}
+
+template<Alphabet alphabet_t>
+void mutate_substitution(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed = 0){
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<uint8_t> dis_alpha_short(0, alphabet_size_v<alphabet_t> - 2);
+    std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(seq) - overlap);
+    alphabet_t & cbase = seq[random_pos(gen)];
+    uint8_t crank = to_rank(cbase);
+    uint8_t rrank = dis_alpha_short(gen);
+    if (rrank >=  crank)
+        ++rrank;
+    cbase.assign_rank(rrank);
+}
+
+template<Alphabet alphabet_t>
+auto generate_reads(std::vector<alphabet_t> & ref,
+                    size_t const number_of_reads,
+                    size_t const read_length,
+                    size_t const simulated_errors,
+                    float const prob_insertion,
+                    float const prob_deletion,
+                    size_t const seed = 0)
+{
+    std::vector<std::vector<alphabet_t> > reads;
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<size_t> seeds (0, SIZE_MAX);
+    std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(ref) - read_length - simulated_errors);
+    std::uniform_real_distribution<double> probability(0.0, 1.0);
+    for (size_t i = 0; i < number_of_reads; ++i){
+        size_t rpos = random_pos(gen);
+        std::vector<alphabet_t> read_tmp{ref.begin() + rpos,
+            ref.begin() + rpos + read_length + simulated_errors};
+        for (size_t j = 0; j < simulated_errors; ++j)
+        {
+            double prob = probability(gen);
+            //Substitution
+            if (prob_insertion + prob_deletion < prob)
+            {
+                mutate_substitution(read_tmp, simulated_errors, seeds(gen));
+            }
+            //Insertion
+            else if (prob_insertion < prob)
+            {
+                mutate_insertion(read_tmp, simulated_errors, seeds(gen));
+            }
+            //Deletion
+            else
+            {
+                mutate_deletion(read_tmp, simulated_errors, seeds(gen));
+            }
+        }
+        read_tmp.erase(read_tmp.begin() + read_length, read_tmp.end());
+        reads.push_back(read_tmp);
+    }
+    return reads;
+}
+
+//============================================================================
+//  undirectional; trivial_search, single, dna4, searched_error = 3
+//============================================================================
+
+void unidirectional_search(benchmark::State & state)
+{
+    uint8_t const simulated_errors = state.range(0);
+    uint8_t const searched_errors = state.range(1);
+    size_t reference_length{100'000};
```

Comment:
You can make all variables `const`.

Is there a reason why you use the assignment operator for `uint8_t`?